### PR TITLE
WT-2994 Create documentation describing page sizes and relationships

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
@daveh86  : Can you please review

**Comment by sulabhM** [comment]
@daveh86 

Here's the final edit of the document that I based my changes on:
https://docs.google.com/document/d/15CascHc4ZblYH-KQAR8zDRAU-AcbpSUwvXJABytrsRc/edit

File: src/docs/tune-page-size-and-comp.dox [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+ - An integer, with acceptable values between 512B and 10TB
+ - Default size: 5 MB
+ - Additionally constrained by the condition:
+   leaf_page_max <= memory_page_max <= cache_size/10
+ - Motivation to tune the value:
+\n memory_page_max is significant for applications wanting to tune for
+consistency in write intensive workloads.
+   - This is the parameter to start with for tuning and trying different values
+to find the correct balance between frequency and duration for an application’s
+needs for each table.
```
Comment (daveh86) [inline src/docs/tune-page-size-and-comp.dox]: Needs for what? It looks like this is covered in some of the dot points below, but reading in order, this dot point feels like it is missing some relevant piece of information.

File: src/docs/tune-page-size-and-comp.dox [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+   - Splitting a growing in-memory page into smaller pages and reconciliation
+both require exclusive access to the page which makes an application's write
+operations wait. Having a large memory_page_max means that the pages will need
+to be split and reconciled less often.  But when that happens, the duration is
```
Comment (daveh86) [inline src/docs/tune-page-size-and-comp.dox]: Duration of what?

File: src/docs/tune-page-size-and-comp.dox [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+WT_SESSION::create(). These options can take values of "english" (to use a
+built-in English language frequency table), "utf8<file>" or "utf16<file>" (to
+use a custom utf8 or utf16 symbol frequency table file). An example of such a
+configuration string is as follows:
+
+<pre>
+          "key_format=S,value_format=S,huffman_key=english,huffman_value=english"
+</pre>
+
+ - Block Compression
```
Comment (daveh86) [inline src/docs/tune-page-size-and-comp.dox]: I think this is a good place to reference tune_compression docs page, as this covers much of the same material. Or you could fold some of the extra compression information in tune_compresson into this page and delete the tune_compresssion page

File: src/docs/tune-page-size-and-comp.dox (hunk 1 above)
Comment (sulabhM) [inline src/docs/tune-page-size-and-comp.dox]: Fixed.

File: src/docs/tune-page-size-and-comp.dox (hunk 2 above)
Comment (sulabhM) [inline src/docs/tune-page-size-and-comp.dox]: Fixed.

File: src/docs/tune-page-size-and-comp.dox (hunk 3 above)
Comment (sulabhM) [inline src/docs/tune-page-size-and-comp.dox]: There wasn't much new information in tune-compression, added a paragraph here and removed  tune-compression.dox

**Comment by daveh86** [comment]
LGTM

**Comment by daveh86** [comment]
@agorrod do you want to take a look at this?

**Comment by agorrod** [comment]
Thanks @sulabhM lgtm, I'll merge.

**Pull request merged**